src/AppInstallerRepositoryCore/PackageTrackingCatalog.cpp

```diff
@@ -31,19 +32,39 @@ namespace AppInstaller::Repository
             return result;
         }
 
+        // Call while holding the CrossProcessLock
+        SQLiteIndex CreateOnlyTrackingIndex(const std::filesystem::path& trackingDB)
+        {
+            return SQLiteIndex::CreateNew(trackingDB.u8string(), SQLite::Version::Latest(), SQLiteIndex::CreateOptions::SupportPathless | SQLiteIndex::CreateOptions::DisableDependenciesSupport);
+        }
+
         // Call while holding the CrossProcessLock
         SQLiteIndex CreateOrOpenTrackingIndex(const std::filesystem::path& trackingDB)
         {
             if (!std::filesystem::exists(trackingDB))
             {
                 std::filesystem::create_directories(trackingDB.parent_path());
-                return SQLiteIndex::CreateNew(trackingDB.u8string(), SQLite::Version::Latest(), SQLiteIndex::CreateOptions::SupportPathless | SQLiteIndex::CreateOptions::DisableDependenciesSupport);
+                return CreateOnlyTrackingIndex(trackingDB);
             }
             else
             {
-                // TODO: Check schema version and upgrade as necessary when there is a relevant new schema.
-                //       Could write this all now but it will be better tested when there is a new schema.
-                return SQLiteIndex::Open(trackingDB.u8string(), SQLiteIndex::OpenDisposition::ReadWrite);
+                try
+                {
+                    // TODO: Check schema version and upgrade as necessary when there is a relevant new schema.
+                    //       Could write this all now but it will be better tested when there is a new schema.
+                    return SQLiteIndex::Open(trackingDB.u8string(), SQLiteIndex::OpenDisposition::ReadWrite);
+                }
+                catch(...)
+                {
+                    LOG_CAUGHT_EXCEPTION_MSG("Exception opening tracking catalog");
+                }
+
+                // Move existing database and create a new one
+                std::filesystem::path destination{ trackingDB };
+                destination.replace_filename(c_PackageTrackingCorruptedFileName);
+                SQLite::SQLiteStorageBase::RenameSQLiteDatabase(trackingDB, destination, true);

```

Comment:
Is there a goal to eventually try to fix or recover some data from corrupted databases?

Reply:
Certainly not programmatically, as the issue here is that SQLite could not open the file as a database (making it quite difficult to operate on) or that we are missing the table that contains the schema version.  The goal of the rename is to allow for human investigation if needed rather than removing any trace of the error.